common/src/main/java/io/github/notenoughupdates/moulconfig/gui/component/TextFieldComponent.kt

```diff
@@ -25,6 +25,7 @@ open class TextFieldComponent(
     private var scrollOffset = 0
     private var visibleText: String? = null
     override fun getWidth(): Int {
+        if (isFocused) return max(preferredWidth, font.getStringWidth(text.get()) + 10)
```

Comment:
I don't think this unilaterally applies to all text fields. Please make this an option instead. Also consider if this not makes more sense to set the width from outside the component instead. (getWidth changes can lead to massive layout changes in more complex layouts.)